oak_functions/client/java/src/com/google/oak/functions/client/AttestationClient.java

```diff
@@ -20,7 +22,6 @@
 import oak.functions.server.RemoteAttestationGrpc;
 import oak.functions.server.RemoteAttestationGrpc.RemoteAttestationStub;
 
-// TODO(#2121): Implement a protocol independent state machine.
```

Comment:
I am not sure this PR actually implements a protocol-agnostic state machine. What are the states and the transitions? I thought the state machine was going to be for the entire attestated invocation process, with states such as `ready to exchange keys` -> ... ->  `ready to accept a request` -> `encrypted request` -> ... -> `Done (response sent)`.

Reply:
Updated.
Now I have `UnattestedPeer` with the `attest` method, that returns `AttestedPeer` with `encrypt`/`decrypt` methods.
Similar to what is being done in https://github.com/project-oak/oak/pull/2156